api/server/router/local/image.go

```diff
@@ -137,6 +137,12 @@ func (s *router) postImagesCreate(ctx context.Context, w http.ResponseWriter, r
 				err = s.daemon.PullImage(ref, metaHeaders, authConfig, output)
 			}
 		}
+		// Check the error from pulling an image to make sure the request
+		// was authorized. Modify the status if the request was
+		// unauthorized to respond with 401 rather than 500.
+		if err != nil && strings.Contains(err.Error(), "unauthorized") {
```

Comment:
I think either of that sounds ideal, I didn't know we already had those constants.